### ISSUE TITLE
Can't open upper case file extension (e.g. .JPEG)

### ISSUE THREAD

**Issue by roaldarbol** [napari/napari#5663]
## 🐛 Bug

Opening a file with upper case file extension throws an error, whereas having the file extension in lower case works.

## To Reproduce

1. Open napari
2. Open image `some_image.JPEG` gives error
3. Change the file extension to `some_image.jpeg`
4. Open image `some_image.jpeg` successfully

<details>
  <summary>Error message</summary>

```bash

ReaderPluginError: Tried to read /Users/roaldarbol/Downloads/cell_microscope071117.JPEG with plugin napari, because it was associated with that file extension/because it is the only plugin capable of reading that path, but it gave an error. Try associating a different plugin or installing a different plugin for this kind of file.
WARNING: Populating font family aliases took 460 ms. Replace uses of missing font family "Monospace" with one that exists to avoid this cost. 
16:42:34 WARNING Populating font family aliases took 460 ms. Replace uses of missing font family "Monospace" with one that exists to avoid this cost. 
(cellpose_napari) ➜  ~ napari                   
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
File /usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/components/viewer_model.py:1126, in ViewerModel._open_or_raise_error(self=Viewer(axes=Axes(visible=False, labels=True, col...ings._transform_active_layer at 0x7ff058a0e310>}), paths=['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'], kwargs={}, layer_type=None, stack=False)
   1125 try:
-> 1126     added = self._add_layers_with_plugins(
        added = []
        self = Viewer(axes=Axes(visible=False, labels=True, colored=True, dashed=False, arrows=True), camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, interactive=True), cursor=Cursor(position=(1.0, 1.0), scaled=True, size=1, style=<CursorStyle.STANDARD: 'standard'>), dims=Dims(ndim=2, ndisplay=2, last_used=0, range=((0, 2, 1), (0, 2, 1)), current_step=(0, 0), order=(0, 1), axis_labels=('0', '1')), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False), layers=[], scale_bar=ScaleBar(visible=False, colored=False, color=<class 'numpy.ndarray'> (4,) float32, ticks=True, position=<Position.BOTTOM_RIGHT: 'bottom_right'>, font_size=10.0, box=False, box_color=<class 'numpy.ndarray'> (4,) float32, unit=None), text_overlay=TextOverlay(visible=False, color=<class 'numpy.ndarray'> (4,) float32, font_size=10.0, position=<TextOverlayPosition.TOP_LEFT: 'top_left'>, text=''), overlays=Overlays(interaction_box=InteractionBox(points=None, show=False, show_handle=False, show_vertices=False, selection_box_drag=None, selection_box_final=None, transform_start=<napari.utils.transforms.transforms.Affine object at 0x7ff058323520>, transform_drag=<napari.utils.transforms.transforms.Affine object at 0x7ff058323580>, transform_final=<napari.utils.transforms.transforms.Affine object at 0x7ff0583235e0>, transform=<napari.utils.transforms.transforms.Affine object at 0x7ff058323640>, allow_new_selection=True, selected_vertex=None)), help='', status='Hold <Alt> key to open plugin selection. Hold <Shift> to open files as stack.', tooltip=Tooltip(visible=True, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[<function InteractionBoxMouseBindings.initialize_mouse_events.<locals>.mouse_move at 0x7ff059111700>], mouse_drag_callbacks=[<function InteractionBoxMouseBindings.initialize_mouse_events.<locals>.mouse_drag at 0x7ff0590ff820>], mouse_double_click_callbacks=[], mouse_wheel_callbacks=[<function dims_scroll at 0x7ff0561d7b80>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, keymap={'Shift': <function InteractionBoxMouseBindings.initialize_key_events.<locals>.hold_to_lock_aspect_ratio at 0x7ff0590ffb80>, 'Control-Shift-R': <function InteractionBoxMouseBindings._reset_active_layer_affine at 0x7ff058a0e280>, 'Control-Shift-A': <function InteractionBoxMouseBindings._transform_active_layer at 0x7ff058a0e310>})
        paths = ['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG']
        kwargs = {}
        stack = False
        plugin = 'napari'
        layer_type = None
   1127         paths,
   1128         kwargs=kwargs,
   1129         stack=stack,
   1130         plugin=plugin,
   1131         layer_type=layer_type,
   1132     )
   1133 # plugin failed

File /usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/components/viewer_model.py:1216, in ViewerModel._add_layers_with_plugins(self=Viewer(axes=Axes(visible=False, labels=True, col...ings._transform_active_layer at 0x7ff058a0e310>}), paths=['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'], stack=False, kwargs={}, plugin='napari', layer_type=None)
   1215     assert len(paths) == 1
-> 1216     layer_data, hookimpl = read_data_with_plugins(
        paths = ['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG']
        stack = False
        plugin = 'napari'
   1217         paths, plugin=plugin, stack=stack
   1218     )
   1220 # glean layer names from filename. These will be used as *fallback*
   1221 # names, if the plugin does not return a name kwarg in their meta dict.

File /usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/plugins/io.py:100, in read_data_with_plugins(paths=['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'], plugin='napari', stack=False)
     95     message = trans._(
     96         'No plugin found capable of reading {repr_path!r}.',
     97         deferred=True,
     98         repr_path=npe1_path,
     99     )
--> 100     raise ValueError(message)
        message = "No plugin found capable of reading '/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'."
    102 if plugin not in plugin_manager.plugins:

ValueError: No plugin found capable of reading '/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'.

The above exception was the direct cause of the following exception:

ReaderPluginError                         Traceback (most recent call last)
File /usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/_qt/qt_viewer.py:830, in QtViewer._qt_open(self=<napari._qt.qt_viewer.QtViewer object>, filenames=['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'], stack=False, choose_plugin=False, plugin=None, layer_type=None, **kwargs={})
    829 try:
--> 830     self.viewer.open(
        self = <napari._qt.qt_viewer.QtViewer object at 0x7ff0583d2ee0>
        self.viewer = Viewer(axes=Axes(visible=False, labels=True, colored=True, dashed=False, arrows=True), camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, interactive=True), cursor=Cursor(position=(1.0, 1.0), scaled=True, size=1, style=<CursorStyle.STANDARD: 'standard'>), dims=Dims(ndim=2, ndisplay=2, last_used=0, range=((0, 2, 1), (0, 2, 1)), current_step=(0, 0), order=(0, 1), axis_labels=('0', '1')), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False), layers=[], scale_bar=ScaleBar(visible=False, colored=False, color=<class 'numpy.ndarray'> (4,) float32, ticks=True, position=<Position.BOTTOM_RIGHT: 'bottom_right'>, font_size=10.0, box=False, box_color=<class 'numpy.ndarray'> (4,) float32, unit=None), text_overlay=TextOverlay(visible=False, color=<class 'numpy.ndarray'> (4,) float32, font_size=10.0, position=<TextOverlayPosition.TOP_LEFT: 'top_left'>, text=''), overlays=Overlays(interaction_box=InteractionBox(points=None, show=False, show_handle=False, show_vertices=False, selection_box_drag=None, selection_box_final=None, transform_start=<napari.utils.transforms.transforms.Affine object at 0x7ff058323520>, transform_drag=<napari.utils.transforms.transforms.Affine object at 0x7ff058323580>, transform_final=<napari.utils.transforms.transforms.Affine object at 0x7ff0583235e0>, transform=<napari.utils.transforms.transforms.Affine object at 0x7ff058323640>, allow_new_selection=True, selected_vertex=None)), help='', status='Hold <Alt> key to open plugin selection. Hold <Shift> to open files as stack.', tooltip=Tooltip(visible=True, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[<function InteractionBoxMouseBindings.initialize_mouse_events.<locals>.mouse_move at 0x7ff059111700>], mouse_drag_callbacks=[<function InteractionBoxMouseBindings.initialize_mouse_events.<locals>.mouse_drag at 0x7ff0590ff820>], mouse_double_click_callbacks=[], mouse_wheel_callbacks=[<function dims_scroll at 0x7ff0561d7b80>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, keymap={'Shift': <function InteractionBoxMouseBindings.initialize_key_events.<locals>.hold_to_lock_aspect_ratio at 0x7ff0590ffb80>, 'Control-Shift-R': <function InteractionBoxMouseBindings._reset_active_layer_affine at 0x7ff058a0e280>, 'Control-Shift-A': <function InteractionBoxMouseBindings._transform_active_layer at 0x7ff058a0e310>})
        filenames = ['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG']
        stack = False
        plugin = None
        layer_type = None
        kwargs = {}
    831         filenames,
    832         stack=stack,
    833         plugin=plugin,
    834         layer_type=layer_type,
    835         **kwargs,
    836     )
    837 except ReaderPluginError as e:

File /usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/components/viewer_model.py:1024, in ViewerModel.open(self=Viewer(axes=Axes(visible=False, labels=True, col...ings._transform_active_layer at 0x7ff058a0e310>}), path=['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'], stack=False, plugin=None, layer_type=None, **kwargs={})
   1022 # no plugin choice was made
   1023 else:
-> 1024     layers = self._open_or_raise_error(
        layers = <module 'napari.layers' from '/usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/layers/__init__.py'>
        self = Viewer(axes=Axes(visible=False, labels=True, colored=True, dashed=False, arrows=True), camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0.0, interactive=True), cursor=Cursor(position=(1.0, 1.0), scaled=True, size=1, style=<CursorStyle.STANDARD: 'standard'>), dims=Dims(ndim=2, ndisplay=2, last_used=0, range=((0, 2, 1), (0, 2, 1)), current_step=(0, 0), order=(0, 1), axis_labels=('0', '1')), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False), layers=[], scale_bar=ScaleBar(visible=False, colored=False, color=<class 'numpy.ndarray'> (4,) float32, ticks=True, position=<Position.BOTTOM_RIGHT: 'bottom_right'>, font_size=10.0, box=False, box_color=<class 'numpy.ndarray'> (4,) float32, unit=None), text_overlay=TextOverlay(visible=False, color=<class 'numpy.ndarray'> (4,) float32, font_size=10.0, position=<TextOverlayPosition.TOP_LEFT: 'top_left'>, text=''), overlays=Overlays(interaction_box=InteractionBox(points=None, show=False, show_handle=False, show_vertices=False, selection_box_drag=None, selection_box_final=None, transform_start=<napari.utils.transforms.transforms.Affine object at 0x7ff058323520>, transform_drag=<napari.utils.transforms.transforms.Affine object at 0x7ff058323580>, transform_final=<napari.utils.transforms.transforms.Affine object at 0x7ff0583235e0>, transform=<napari.utils.transforms.transforms.Affine object at 0x7ff058323640>, allow_new_selection=True, selected_vertex=None)), help='', status='Hold <Alt> key to open plugin selection. Hold <Shift> to open files as stack.', tooltip=Tooltip(visible=True, text=''), theme='dark', title='napari', mouse_over_canvas=False, mouse_move_callbacks=[<function InteractionBoxMouseBindings.initialize_mouse_events.<locals>.mouse_move at 0x7ff059111700>], mouse_drag_callbacks=[<function InteractionBoxMouseBindings.initialize_mouse_events.<locals>.mouse_drag at 0x7ff0590ff820>], mouse_double_click_callbacks=[], mouse_wheel_callbacks=[<function dims_scroll at 0x7ff0561d7b80>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, keymap={'Shift': <function InteractionBoxMouseBindings.initialize_key_events.<locals>.hold_to_lock_aspect_ratio at 0x7ff0590ffb80>, 'Control-Shift-R': <function InteractionBoxMouseBindings._reset_active_layer_affine at 0x7ff058a0e280>, 'Control-Shift-A': <function InteractionBoxMouseBindings._transform_active_layer at 0x7ff058a0e310>})
        _path = ['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG']
        _stack = False
        kwargs = {}
        layer_type = None
   1025         _path, kwargs, layer_type, _stack
   1026     )
   1027     added.extend(layers)

File /usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/components/viewer_model.py:1135, in ViewerModel._open_or_raise_error(self=Viewer(axes=Axes(visible=False, labels=True, col...ings._transform_active_layer at 0x7ff058a0e310>}), paths=['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'], kwargs={}, layer_type=None, stack=False)
   1134     except Exception as e:
-> 1135         raise ReaderPluginError(
        trans = <napari.utils.translations.TranslationBundle object at 0x7ff05267df10>
        plugin = 'napari'
        paths = ['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG']
   1136             trans._(
   1137                 'Tried opening with {plugin}, but failed.',
   1138                 deferred=True,
   1139                 plugin=plugin,
   1140             ),
   1141             plugin,
   1142             paths,
   1143         ) from e
   1144 # multiple plugins
   1145 else:

ReaderPluginError: Tried opening with napari, but failed.

The above exception was the direct cause of the following exception:

ReaderPluginError                         Traceback (most recent call last)
File /usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/_qt/qt_viewer.py:1191, in QtViewer.dropEvent(self=<napari._qt.qt_viewer.QtViewer object>, event=<PyQt5.QtGui.QDropEvent object>)
   1188     else:
   1189         filenames.append(url.toString())
-> 1191 self._qt_open(
        self = <napari._qt.qt_viewer.QtViewer object at 0x7ff0583d2ee0>
        filenames = ['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG']
        shift_down = <PyQt5.QtCore.Qt.KeyboardModifiers object at 0x7ff03c43bba0>
        alt_down = <PyQt5.QtCore.Qt.KeyboardModifiers object at 0x7ff03c43bb30>
   1192     filenames,
   1193     stack=bool(shift_down),
   1194     choose_plugin=bool(alt_down),
   1195 )

File /usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/_qt/qt_viewer.py:838, in QtViewer._qt_open(self=<napari._qt.qt_viewer.QtViewer object>, filenames=['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'], stack=False, choose_plugin=False, plugin=None, layer_type=None, **kwargs={})
    830     self.viewer.open(
    831         filenames,
    832         stack=stack,
   (...)
    835         **kwargs,
    836     )
    837 except ReaderPluginError as e:
--> 838     handle_gui_reading(
        filenames = ['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG']
        self = <napari._qt.qt_viewer.QtViewer object at 0x7ff0583d2ee0>
        stack = False
        layer_type = None
        kwargs = {}
    839         filenames,
    840         self,
    841         stack,
    842         e.reader_plugin,
    843         e,
    844         layer_type=layer_type,
    845         **kwargs,
    846     )
    847 except MultipleReaderError:
    848     handle_gui_reading(filenames, self, stack, **kwargs)

File /usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/_qt/dialogs/qt_reader_dialog.py:188, in handle_gui_reading(paths=['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'], qt_viewer=<napari._qt.qt_viewer.QtViewer object>, stack=False, plugin_name='napari', error=ReaderPluginError('Tried opening with napari, but failed.'), plugin_override=False, **kwargs={'layer_type': None})
    161 """Present reader dialog to choose reader and open paths based on result.
    162 
    163 This function is called whenever ViewerModel._open_or_get_error returns
   (...)
    185     Dictates whether checkbox to remember choice is unchecked by default
    186 """
    187 _path = paths[0]
--> 188 readers = prepare_remaining_readers(paths, plugin_name, error)
        paths = ['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG']
        plugin_name = 'napari'
        error = ReaderPluginError('Tried opening with napari, but failed.')
    189 error_message = str(error) if error else ''
    190 readerDialog = QtReaderDialog(
    191     parent=qt_viewer,
    192     pth=_path,
   (...)
    195     persist_checked=not plugin_override,
    196 )

File /usr/local/anaconda3/envs/cellpose_napari/lib/python3.8/site-packages/napari/_qt/dialogs/qt_reader_dialog.py:243, in prepare_remaining_readers(paths=['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'], plugin_name='napari', error=ReaderPluginError('Tried opening with napari, but failed.'))
    241 # if there's no other readers left, raise the exception
    242 if not readers and error:
--> 243     raise ReaderPluginError(
        trans = <napari.utils.translations.TranslationBundle object at 0x7ff05267df10>
        paths = ['/Users/roaldarbol/Downloads/cell_microscope071117.JPEG']
        len(paths) > 1 = False
        len(paths) = 1
        paths[0] = '/Users/roaldarbol/Downloads/cell_microscope071117.JPEG'
        plugin_name = 'napari'
        error = ReaderPluginError('Tried opening with napari, but failed.')
    244         trans._(
    245             "Tried to read {path_message} with plugin {plugin}, because it was associated with that file extension/because it is the only plugin capable of reading that path, but it gave an error. Try associating a different plugin or installing a different plugin for this kind of file.",
    246             path_message=f"[{paths[0]}, ...]"
    247             if len(paths) > 1
    248             else paths[0],
    249             plugin=plugin_name,
    250         ),
    251         plugin_name,
    252         paths,
    253     ) from error
    255 return readers

ReaderPluginError: Tried to read /Users/roaldarbol/Downloads/cell_microscope071117.JPEG with plugin napari, because it was associated with that file extension/because it is the only plugin capable of reading that path, but it gave an error. Try associating a different plugin or installing a different plugin for this kind of file.
```

</details>

## Expected behavior

As many DSLR cameras output images upper case file names, I'd expect there to be no difference between lower and upper case file extensions.

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

```
napari: 0.4.17
Platform: macOS-10.16-x86_64-i386-64bit
System: MacOS 12.6.3
Python: 3.8.16 (default, Mar 1 2023, 21:19:10) [Clang 14.0.6 ]
Qt: 5.15.2
PyQt5: 5.15.9
NumPy: 1.23.5
SciPy: 1.9.3
Dask: 2023.2.0
VisPy: 0.11.0
magicgui: 0.7.2
superqt: unknown
in-n-out: 0.1.7
app-model: 0.1.2
npe2: 0.6.2

OpenGL:
- GL version: 2.1 INTEL-18.8.6
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1440x900, scale 2.0
- screen 2: resolution 1680x1050, scale 1.0
```

 - Any other relevant information:

**Comment by Czaki** [comment]
It was initially reported here #4986 and looks like #5121 does not solve the problem.

**Comment by psobolewskiPhD** [comment]
Wow, this is a bit embarrassing and a serious issue!
I have numerous files with all caps extensions, I guess i've never used them with napari.
So the other PR ensures a plugin is called, so so I think the issue is in npe2—here's where's I've traced this:
https://github.com/napari/napari/blob/b1d7c4098287a83f5ecb3e5de6756fb0264cbd8c/napari/plugins/_npe2.py#L54-L58
napari calls npe2 here.
In npe2 I've tracked it down to here:

https://github.com/napari/npe2/blob/d7329b7088f59ac743a67ff47aca32067b5a246a/npe2/_plugin_manager.py#L145
Edit: no idea why it's not showing, here's the line:
```
            yield from {r for pattern, r in self._readers if fnmatch(path, pattern)}
```

`fnmatch` is case-sensitive on non-windows, so the path is compared to the `pattern` which is from the manifest--all lower case.

So our options are: 1) pass `.lower()` in napari when calling the npe2 `read_get_reader` or put `.lower()` in that `fnmatch` in npe2.
I can make a PR for either, but don't have good intuition which is better.
cc: @jni @DragaDoncila @Czaki
